.github/actions/setup-guardian/action.yml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Guardian'
+
+description: |-
+  Use this action to setup the Guardian binary for use within a GitHub workflow.
+
+inputs:
+  guardian_version:
+    description: 'The version of the Guardian CLI to install.'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Cache Guardian'
```

Comment:
Is using the cache really worth it? You are just downloading and untarballing the binary - its the same amount of work as the cache AFAIK (cache downloads a tarball and untarballs it).

Reply:
This is a common practice for actions, e.g. setup-go, setup-node, etc. I believe you are correct in the process of how the cache works, but my assumption was there less latency or something since its "local to GitHub" (I might be just wishing here).

Also, once Guardian is cached on the main branch it can then be used on feature branches in pull requests and for applies. Since we require guardian two times for plan and apply (to init), I think this makes most sense.

Reply:
Ack. This makes sense in theory and I see no reason not to do it as either the cache should be more performant (or likely eventually will be).

But I still honestly would be curious to see a test to compare the performance of both.

Reply:
You're both correct. One thing I'll note: if we cache the download, getting future usage metrics can be more challenging. If we don't cache, each download corresponds to a unique run.